domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputHasNumeratorEqualToRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,203 @@
+package org.oppia.android.domain.classify.rules.fractioninput
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.Fraction
+import org.oppia.android.app.model.InteractionObject
+import org.oppia.android.domain.classify.RuleClassifier
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KClass
+import kotlin.reflect.full.cast
+import kotlin.test.fail
+
+/** Tests for [FractionInputHasNumeratorEqualToRuleClassifierProvider]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(manifest = Config.NONE)
+class FractionInputHasNumeratorEqualToRuleClassifierProviderTest {
+  private val NON_NEGATIVE_VALUE_0 = createNonNegativeInt(value = 0)
+  private val WHOLE_NUMBER_123 = createWholeNumber(isNegative = false, value = 123)
+  private val FRACTION_2_OVER_4 = createFraction(isNegative = false, numerator = 2, denominator = 4)
+  private val SIGNED_INT_1 = createSignedInt(1)
+  private val SIGNED_INT_2 = createSignedInt(2)
+  private val SIGNED_INT_NEGATIVE_2 = createSignedInt(-2)
+
+  @Inject
+  internal lateinit var fractionInputHasNumeratorEqualToRuleClassifierProvider:
+    FractionInputHasNumeratorEqualToRuleClassifierProvider
+
+  private val numeratorIsEqualClassifierProvider: RuleClassifier by lazy {
+    fractionInputHasNumeratorEqualToRuleClassifierProvider.createRuleClassifier()
+  }
+
+  @Test
+  fun testEquals_wholeNumber123Answer_withSignedInt1Input_bothValuesDoNotMatch() {
```

Comment:
Here & elsewhere: aren't we testing numerator equals not general equals? Maybe: ``testNumeratorEquals_...`` here & elsewhere would be clearer.